src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -33,23 +33,20 @@ internal sealed partial class BoundLocalFunctionStatement : IBoundLambdaOrFuncti
         BoundBlock IBoundLambdaOrFunction.Body { get => this.Body; }
     }
 
-    internal struct InferredLambdaReturnType
+    internal readonly struct InferredLambdaReturnType
     {
-        internal readonly int NumExpressions;
-        internal readonly bool HadExpressionlessReturn;
+        internal readonly bool HasReturnValue;
         internal readonly RefKind RefKind;
         internal readonly TypeWithAnnotations TypeWithAnnotations;
         internal readonly ImmutableArray<DiagnosticInfo> UseSiteDiagnostics;
 
         internal InferredLambdaReturnType(
-            int numExpressions,
-            bool hadExpressionlessReturn,
+            bool hasReturnValue,
             RefKind refKind,
             TypeWithAnnotations typeWithAnnotations,
             ImmutableArray<DiagnosticInfo> useSiteDiagnostics)
         {
-            NumExpressions = numExpressions;
-            HadExpressionlessReturn = hadExpressionlessReturn;
+            HasReturnValue = hasReturnValue;
```

Comment:
>HasReturnValue = hasReturnValue; [](start = 12, length = 32)

It looks like this refactoring is not related to primary goal of this PR, please exclude it from the PR.